### PR TITLE
ref(maven): Improve logging for `maven` target

### PR DESCRIPTION
**Pull request by iker-barriocanal**
- Don't log when not finding BOMs. Most of the packages aren't supposed to be BOM files, and logging this may give the impression that something is missing when it's not.
- Add a note to the target's docs to show command output when the log level is `trace`.

### CODE REVIEW

File: src/targets/maven.ts [hunk 1]
```diff
@@ -411,4 +412,14 @@ export class MavenTarget extends BaseTarget {
         )
       : `${moduleName}.jar`;
   }
+
+  private logCmdOutput(output: Buffer | undefined): void {
+    if (output) {
+      if (output.length === 0) {
```
Comment (AbhiPrasad) [inline src/targets/maven.ts:418]: We can combine this conditional.

File: src/targets/maven.ts [hunk 2]
```diff
@@ -411,4 +412,14 @@ export class MavenTarget extends BaseTarget {
...
+        this.logger.debug(`The command didn't have any output.`);
```
Comment (AbhiPrasad) [inline src/targets/maven.ts:419]: Hard to know what the command is from this statement, would be helpful to have some details about the command

File: src/targets/maven.ts [hunk 3]
```diff
@@ -411,4 +412,14 @@ export class MavenTarget extends BaseTarget {
         )
       : `${moduleName}.jar`;
   }
+
+  private logCmdOutput(output: Buffer | undefined): void {
```
Comment (BYK) [inline src/targets/maven.ts:416]: You shouldn't need any of this as `spawnProcess` does it underneath:

https://github.com/getsentry/craft/blob/a015c79de78bbc672570a38cb2177786e2fa93db/src/utils/system.ts#L174-L182

**Review by BYK — CHANGES_REQUESTED** [review]

File: src/targets/maven.ts [hunk 4]
```diff
@@ -331,16 +332,20 @@ export class MavenTarget extends BaseTarget {
...
+        'gpg:sign-and-deploy-file',
+        `-Dfile=${bomFile}`,
+        `-DpomFile=${bomFile}`,
+        `-DrepositoryId=${this.mavenConfig.mavenRepoId}`,
+        `-Durl=${this.mavenConfig.mavenRepoUrl}`,
+        '--settings',
+        this.mavenConfig.mavenSettingsPath,
+      ],
+      {},
+      { showStdout: this.logger.level >= LogLevel.Debug }
```
Comment (BYK) [inline src/targets/maven.ts:347]: Why are we doing this?

**Review by BYK — CHANGES_REQUESTED** [review]
I don't understand the motivation behind the `this.logger.level >= LogLevel.Debug` condition.

`spawnProcess` already logs with level `info` if this is `true` at all times. If we want to see the output for debugging purposes, we can always set the logging level to `trace` for that specific session, right?

**Review by BYK — APPROVED** [review]

**Pull request merged**